src/core/hyper-ref/index.js

```diff
@@ -62,6 +62,8 @@ export default class HyperRef extends PureComponent<Props, State> {
     refreshing: false,
   };
 
+  behaviorElements: Element[];
```

Comment:
This makes me a little nervous, I don't remember when all the React component lifecycle methods fire. Will `componentDidUpdate` always get called, including when we first render the component?

Reply:
`componentDidUpdate` only gets called when the state/props change (see [React doc](https://reactjs.org/docs/react-component.html#componentdidupdate)). For the initial rendering, we rely on setting these in the component's constructor

Reply:
Ah I see that was added in a later commit, thanks.